File: 프로그래머스/2/42860. 조이스틱/조이스틱.py
def solution(name):
    answer = 0
    min_move = len(name)-1

    for i, k in enumerate(name):
        answer += min(ord(k) - ord('A'), ord('Z')-ord(k)+1)

        next = i+1
        while next<len(name) and name[next] == 'A':
            next+=1

        min_move = min(min_move, 2*i +len(name)-next, i+ 2*(len(name)-next))

    answer += min_move
    return answer

File: 프로그래머스/2/42860. 조이스틱/조이스틱.java
class Solution {
    public static int solution(String name) {
        int answer = 0;
        int n = name.length();
        int min_move = n - 1;

        for(int i=0; i<n; i++){
            char c = name.charAt(i);
            answer += Math.min(c-'A','Z'-c+1);

            int next = i+1;
            while(next<n && name.charAt(next) == 'A'){
                next += 1;
            }

            min_move = Math.min(min_move, Math.min(2*i+n-next, i+2*(n-next)));
        }

        answer += min_move;
        return answer;
    }
}